### PR TITLE
astrocalc

### PR DESCRIPTION
**Pull request by nxdefiant**
- Compatibility with Bangle.js 2
- Get location from My Location

### CODE REVIEW

File: modules/suncalc.js [hunk 1]
```diff
@@ -279,7 +279,7 @@ function hoursLater(date, h) {
 // calculations for moon rise/set times are based on http://www.stargazing.net/kepler/moonrise.html article
 
 SunCalc.getMoonTimes = function (date, lat, lng, inUTC) {
-    var t = new Date(date);
+    var t = typeof(date) === "object" ? date : new Date(date);
```
Comment (nxdefiant) [inline modules/suncalc.js:282]: Espruino does not seem to support the
new Date(object) constructor

**Comment by nxdefiant** [comment]
Notify @paulcockrell

**Comment by paulcockrell** [comment]
@nxdefiant looks good but what is setting the GPS coordinates in the mylocation.json?

**Comment by nxdefiant** [comment]
@paulcockrell You go to settings->Apps->My Location and set them there, see https://banglejs.com/apps/?id=mylocation

File: modules/suncalc.js (hunk 1 above)
Comment (gfwilliams) [reply modules/suncalc.js:282]: Thanks for spotting that! I thought I'd tried to diff most of the clones of this file to see if there were differences, but I must have missed this one

**Comment by gfwilliams** [comment]
Thanks for your work on this!

**Pull request merged**